before_after_filesystem_snapshot.py

```diff
@@ -0,0 +1,45 @@
+def before_after_snapshot(before_dict, after_dict):
+  '''before_after_snapshot is a simple function that outputs which files were
+    unchanged, modified, added or removed after a step was carried out.'''
```

Comment:
I generally recommend to stay within the scope of the function, when writing a docstring. A reader might not know what "a step was carried out means", and also does not need to. I, as a user of this function, would rather like to know what the arguments need to look like. :)

That said, you can still mention in-toto steps to add some context or an example use case.